src/resources/parser/glb-parser.js

```diff
@@ -298,24 +388,120 @@ Object.assign(pc, function () {
 
     var tempMat = new pc.Mat4();
     var tempVec = new pc.Vec3();
+
     var createMesh = function (device, meshData, accessors, bufferViews, buffers) {
         var meshes = [];
 
+        var semanticMap = {
+            'POSITION': { semantic: pc.SEMANTIC_POSITION,       storageType: pc.TYPE_FLOAT32, byteSize: 4 },
+            'NORMAL': { semantic: pc.SEMANTIC_NORMAL,           storageType: pc.TYPE_FLOAT32, byteSize: 4 },
+            'TANGENT': { semantic: pc.SEMANTIC_TANGENT,         storageType: pc.TYPE_FLOAT32, byteSize: 4 },
+            'BINORMAL': { semantic: pc.SEMANTIC_BINORMAL,       storageType: pc.TYPE_FLOAT32, byteSize: 4 },
+            'COLOR_0': { semantic: pc.SEMANTIC_COLOR,           storageType: pc.TYPE_UINT8,   byteSize: 1,  normalize: true },
+            'JOINTS_0': { semantic: pc.SEMANTIC_BLENDINDICES,   storageType: pc.TYPE_UINT8,   byteSize: 1 },
+            'WEIGHTS_0': { semantic: pc.SEMANTIC_BLENDWEIGHT,   storageType: pc.TYPE_FLOAT32, byteSize: 4 },
+            'TEXCOORD_0': { semantic: pc.SEMANTIC_TEXCOORD0,    storageType: pc.TYPE_FLOAT32, byteSize: 4 },
+            'TEXCOORD_1': { semantic: pc.SEMANTIC_TEXCOORD1,    storageType: pc.TYPE_FLOAT32, byteSize: 4 }
+        };
+
         meshData.primitives.forEach(function (primitive) {
-            var attributes = primitive.attributes;
 
-            // get indices
-            var indices =
-                primitive.hasOwnProperty('indices') ?
-                    getAccessorData(accessors[primitive.indices], bufferViews, buffers) : null;
+            var primitiveType, vertexBuffer, numIndices;
+            var indices = null;
+            var mesh = new pc.Mesh();
 
-            var vertexBuffer = createVertexBuffer(device, attributes, indices, accessors, bufferViews, buffers);
+            // try and get draco compressed data first
+            if (primitive.hasOwnProperty('extensions')) {
+                var extensions = primitive.extensions;
+                if (extensions.hasOwnProperty('KHR_draco_mesh_compression')) {
+
+                    // access DracoDecoderModule
+                    var decoderModule = window.DracoDecoderModule;
+                    if (decoderModule) {
+                        var extDraco = extensions.KHR_draco_mesh_compression;
+                        if (extDraco.hasOwnProperty('attributes')) {
+                            var bufferView = bufferViews[extDraco.bufferView];
+                            var arrayBuffer = buffers[bufferView.buffer];
+                            var uint8Buffer = new Uint8Array(arrayBuffer.buffer, arrayBuffer.byteOffset + bufferView.byteOffset, bufferView.byteLength);
+                            var buffer = new decoderModule.DecoderBuffer();
+                            buffer.Init(uint8Buffer, uint8Buffer.length);
+
+                            var decoder = new decoderModule.Decoder();
```

Comment:
how slow is it to construct a decoder instance? would there be any benefit in caching it?